src/type-layout.md

```diff
@@ -162,7 +163,24 @@ representation will not change the layout of `Inner`.
 Nominal types without a `repr` attribute have the default representation.
 Informally, this representation is also called the `rust` representation.
 
-There are no guarantees of data layout made by this representation.
+There are very few data layout guarantees made by this representation. The only
+guarantees are:
+
+ 1. The fields of the struct are properly aligned.
+ 2. The fields do not overlap.
+ 3. The alignment of the struct is not less than the alignment of any of its
+    fields.
```

Comment:
Should we guarantee that the alignment is equal to the largest alignment of the fields?

Reply:
If it ever becomes possible for PGO to change the layout of rust structs, it may be beneficial to increase alignment for certain types in some cases.

Reply:
(Can be ignored by author of PR)

Is it within the scope of the reference to include non-normative statements like: 

"In practice, there's no reason a compiler would ever require a type to be more aligned than this, but we would like to reserve the right to do so, in case such a situation is found." (Possibly also noting that there isn't any clear reason why a user would *need* to care about this unless they were doing something that probably motivates using other repr annotations anyway, so it's basically just erring on the side of compiler freedom in a situation that doesn't much matter either way.).

Reply:
I would avoid saying anything about an upperbound on alignment for repr(Rust), especially if that statement isn't normative, because it might confuse what *is* guaranteed.

Reply:
Or, for that case, align(16) on x86_64, so you can use aligned SSE reads (`movaps`) to copy it.